src/provider/mod.rs

```diff
@@ -816,34 +824,29 @@ async fn transfer_collection(
     }
 
     debug!("done writing");
-    writer.finish().await?;
+    writer.inner.finish().await?;
     Ok(SentStatus::Sent)
 }
 
-fn notify_transfer_aborted(events: broadcast::Sender<Event>, connection_id: u64, request_id: u64) {
-    let _ = events.send(Event::TransferAborted {
-        connection_id,
-        request_id,
+fn notify_transfer_aborted(writer: &ResponseWriter) {
```

Comment:
we could make this a fn of ResponseWriter. Not sure. ResonseWriter is intended as just a bunch of stuff without any encapsulation, so I think it does not matter that much...